server/sockets/list.js

```diff
@@ -256,12 +256,16 @@ const addMemberRoleInList = (socket, clients) => {
       });
 
     if (clients.has(userId)) {
-      socket.broadcast
-        .to(clients.get(userId))
-        .emit(ListActionTypes.ADD_MEMBER_ROLE_SUCCESS, {
-          ...data,
-          isCurrentUserRoleChanging: true
-        });
+      const { viewId, socketId } = clients.get(userId);
+
+      if (viewId === listId) {
+        socket.broadcast
+          .to(socketId)
+          .emit(ListActionTypes.ADD_MEMBER_ROLE_SUCCESS, {
+            ...data,
+            isCurrentUserRoleChanging: true
+          });
+      }
```

Comment:
same here 
```
  const { viewId, socketId } = clients.get(userId);

if(clients.has(userid) && viewId === listId) { 
    socket......
```

Reply:
we have to first check if exist then destructure to certain props as we discussed